DPL: print an error when the configuration is not parsed correctly

ktf opened the pull request:
This should probably be a fatal error not sure why the rethrown
exception is ignored (and where).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 05bb51b0fabbea0a4c5c667abfeec291e84b0a0e at 2025-10-09 11:44:
```
[2991/4967] Building CXX object Detectors/MUON/Common/CMakeFiles/O2exe-mid-dcs-ccdb.dir/src/dcs-ccdb.cxx.o
[2992/4967] Linking CXX executable stage/bin/o2-calibration-mft-calib-workflow
[2993/4967] Building CXX object Detectors/MUON/MCH/Calibration/CMakeFiles/O2exe-mch-pedestal-decoding-workflow.dir/src/pedestal-decoding-workflow.cxx.o
[2994/4967] Linking CXX executable stage/bin/o2-zdc-digi2raw
[2995/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2lib-FV0Workflow.dir/src/RecoWorkflow.cxx.o
[2996/4967] Building CXX object Detectors/MUON/MCH/Calibration/CMakeFiles/O2exe-calibration-mch-badchannel-calib-workflow.dir/src/badchannel-calib-workflow.cxx.o
[2997/4967] Building CXX object Detectors/MUON/MCH/Calibration/CMakeFiles/O2lib-MCHCalibration.dir/G__O2MCHCalibration.cxx.o
[2998/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/DeliveryType.cxx.o
[2999/4967] Building CXX object Detectors/FIT/FT0/reconstruction/CMakeFiles/O2lib-FT0Reconstruction.dir/src/CTFCoder.cxx.o
[3000/4967] Building CXX object Detectors/MUON/MID/Calibration/CMakeFiles/O2lib-MIDCalibration.dir/G__O2MIDCalibration.cxx.o
[3001/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/GenericFunctions.cxx.o
[3002/4967] Linking CXX shared library stage/lib64/libO2MCHCalibration.so
[3003/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/DataPointIdentifier.cxx.o
[3004/4967] Linking CXX shared library stage/lib64/libO2FT0Reconstruction.so
[3005/4967] Linking CXX shared library stage/lib64/libO2MIDCalibration.so
[3006/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/DataPointCreator.cxx.o
[3007/4967] Generating G__O2ZDCRaw.cxx, ../../../stage/lib64/G__O2ZDCRaw_rdict.pcm, ../../../stage/lib64/libO2ZDCRaw.rootmap
[3008/4967] Generating G__O2ZDCReconstruction.cxx, ../../../stage/lib64/G__O2ZDCReconstruction_rdict.pcm, ../../../stage/lib64/libO2ZDCReconstruction.rootmap
[3009/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/DataPointValue.cxx.o
[3010/4967] Linking CXX executable stage/bin/o2-mch-pedestal-decoding-workflow
[3011/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2lib-FV0Workflow.dir/src/RecPointReaderSpec.cxx.o
[3012/4967] Linking CXX executable stage/bin/o2-calibration-mch-badchannel-calib-workflow
[3013/4967] Linking CXX executable stage/bin/o2-ft0-test-raw-conversion
[3014/4967] Linking CXX executable stage/bin/o2-ft0-test-raw2digit
[3015/4967] Linking CXX executable stage/tests/o2-test-mch-pedestal-data-iterator
[3016/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2lib-FV0Workflow.dir/src/DigitReaderSpec.cxx.o
[3017/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/Clock.cxx.o
[3018/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/StringUtils.cxx.o
[3019/4967] Building CXX object Detectors/DCS/CMakeFiles/O2lib-DetectorsDCS.dir/src/DataPointGenerator.cxx.o
[3020/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/ZDCEnergyParam.cxx.o
[3021/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/ZDCTDCParam.cxx.o
[3022/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/ZDCTowerParam.cxx.o
[3023/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-reco-workflow.dir/src/fv0-reco-workflow.cxx.o
[3024/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-digits-writer-workflow.dir/src/digits-writer-workflow.cxx.o
[3025/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/BaselineParam.cxx.o
[3026/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/RecoParamZDC.cxx.o
[3027/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/NoiseParam.cxx.o
[3028/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-recpoints-writer-workflow.dir/src/recpoints-writer-workflow.cxx.o
[3029/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-digit-reader-workflow.dir/src/digits-reader-workflow.cxx.o
[3030/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/CTFHelper.cxx.o
[3031/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2lib-FV0Workflow.dir/src/ReconstructionSpec.cxx.o
[3032/4967] Linking CXX executable stage/bin/o2-mid-bad-channels-ccdb
[3033/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/DigiParser.cxx.o
[3034/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-entropy-encoder-workflow.dir/src/entropy-encoder-workflow.cxx.o
[3035/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-recpoints-reader-workflow.dir/src/recpoints-reader-workflow.cxx.o
[3036/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2lib-FV0Workflow.dir/src/RecPointWriterSpec.cxx.o
[3037/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/src/ZDCTDCCorr.cxx.o
[3038/4967] Building CXX object Detectors/ZDC/raw/CMakeFiles/O2lib-ZDCRaw.dir/G__O2ZDCRaw.cxx.o
[3039/4967] Building CXX object Detectors/ZDC/reconstruction/CMakeFiles/O2lib-ZDCReconstruction.dir/G__O2ZDCReconstruction.cxx.o
[3040/4967] Building CXX object Detectors/FIT/FV0/workflow/CMakeFiles/O2exe-fv0-integrate-cluster-reader-workflow.dir/src/cluster-integrator-reader.cxx.o
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14727/05bb51b0fabbea0a4c5c667abfeec291e84b0a0e/build_O2_fullCI_slc9/pretty.html).

ktf committed:
DPL: print an error when the configuration is not parsed correctly

This should probably be a fatal error not sure why the rethrown
exception is ignored (and where).
